src/main/java/org/wso2/carbon/connector/utils/Utils.java

```diff
@@ -113,23 +117,37 @@ public static String getTenantSpecificConnectionName(String connectionName, Mess
      * @param diskShareAccessMask Disk share access mask
      * @return Validated disk share access mask
      */
-    public static String validateAndGetDiskShareAccessMask(String diskShareAccessMask) {
-        // Define allowed values (case-insensitive)
-        Set<String> allowedValues = new HashSet<>();
-        allowedValues.add("READ");
-        allowedValues.add("WRITE");
-        allowedValues.add("READ_WRITE");
-        allowedValues.add("ALL");
-        allowedValues.add("MAXIMUM");
-
-        // Validate input
-        if (diskShareAccessMask == null || !allowedValues.contains(diskShareAccessMask.toUpperCase())) {
-            //set disk share access mask to max allowed to keep default behaviour
-            return "MAXIMUM";
+    public static ArrayList<String> validateAndGetDiskShareAccessMask(String diskShareAccessMask) {
+        // Prepare the set of allowed access mask values
+        Set<String> allowedValues = new HashSet<String>();
+        for (AccessMask mask : AccessMask.values()) {
+            allowedValues.add(mask.name());
         }
 
-        // Assign the validated value
-        return diskShareAccessMask.toUpperCase();
+        ArrayList<String> outDiskShareAccessMasks = new ArrayList<String>();
+
+        // Validate and collect allowed values
+        if (diskShareAccessMask != null) {
+            String[] masks = diskShareAccessMask.split(",");
+            for (String mask : masks) {
+                String accessMask = mask.trim();
```

Comment:
String accessMask = mask.trim().toUpperCase(); shall we use this to allow mor flexibility?